### PR TITLE
Remove test coverage of FeatureSIMD=0

### LINKED ISSUE #66565: Test failure JIT\\Regression\\JitBlue\\GitHub_35821\\GitHub_35821\\GitHub_35821.cmd
Run: [runtime-coreclr jitstress-isas-arm 20220312.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1660460&view=ms.vss-test-web.build-test-results-tab&runId=45714182&resultId=101538&paneView=debug)

Failed test:
```
CoreCLR windows arm64 Checked jitstress_isas_nohwintrinsic_nosimd @ Windows.10.Arm64v8.Open

- JIT\\Regression\\JitBlue\\GitHub_35821\\GitHub_35821\\GitHub_35821.cmd

CoreCLR windows arm64 Checked jitstress_isas_nohwintrinsic @ Windows.10.Arm64v8.Open

- JIT\\Regression\\JitBlue\\GitHub_35821\\GitHub_35821\\GitHub_35821.cmd

CoreCLR Linux arm64 Checked jitstress_isas_nohwintrinsic_nosimd @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-20210531091519-97d8652

- JIT/Regression/JitBlue/GitHub_35821/GitHub_35821/GitHub_35821.sh

CoreCLR Linux arm64 Checked jitstress_isas_nohwintrinsic @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-20210531091519-97d8652

- JIT/Regression/JitBlue/GitHub_35821/GitHub_35821/GitHub_35821.sh

CoreCLR OSX arm64 Checked jitstress_isas_nohwintrinsic @ OSX.1200.ARM64.Open

- JIT/Regression/JitBlue/GitHub_35821/GitHub_35821/GitHub_35821.sh

CoreCLR OSX arm64 Checked jitstress_isas_nohwintrinsic_nosimd @ OSX.1200.ARM64.Open

- JIT/Regression/JitBlue/GitHub_35821/GitHub_35821/GitHub_35821.sh

```
**Error message:**
```
Assert failure(PID 7376 [0x00001cd0], Thread: 13288 [0x33e8]): Assertion failed '(allRegs(theInterval->registerType) & mask) != 0' in 'projs.GitHub_35821:Main(System.String[]):int' during 'Linear scan register alloc' (IL size 48; hash 0x3767a9b9; FullOpts)

File: D:\a\_work\1\s\src\coreclr\jit\lsrabuild.cpp Line: 596
Image: D:\h\w\99660882\p\corerun.exe
Return code:      1
Raw output file:      D:\h\w\99660882\w\AA1D09D7\uploads\Reports\JIT.Regression\JitBlue\GitHub_35821\GitHub_35821\GitHub_35821.output.txt
Raw output:
BEGIN EXECUTION
"D:\h\w\99660882\p\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false"  GitHub_35821.dll
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=D:\h\w\99660882\p
D:\h\w\99660882\w\AA1D09D7\e\JIT\Regression\JitBlue\GitHub_35821\GitHub_35821\GitHub_35821.cmd
Expected: True
Actual:   False


Stack trace
   at JIT_Regression._JitBlue_GitHub_35821_GitHub_35821_GitHub_35821_._JitBlue_GitHub_35821_GitHub_35821_GitHub_35821_cmd()
```
or
```
Assert failure(PID 815 [0x0000032f], Thread: 815 [0x032f]): Assertion failed '(allRegs(theInterval->registerType) & mask) != 0' in 'projs.GitHub_35821:Main(System.String[]):int' during 'Linear scan register alloc' (IL size 48; hash 0x3767a9b9; FullOpts)

File: /__w/1/s/src/coreclr/jit/lsrabuild.cpp Line: 596
Image: /root/helix/work/correlation/corerun
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7faba09ff8)
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7faba09ff8)
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7fab9ff070)
/root/helix/work/workitem/e/JIT/Regression/JitBlue/GitHub_35821/GitHub_35821/GitHub_35821.sh: line 370:   815 Aborted                 (core dumped) $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code:      1
Raw output file:      /root/helix/work/workitem/uploads/Reports/JIT.Regression/JitBlue/GitHub_35821/GitHub_35821/GitHub_35821.output.txt
Raw output:
BEGIN EXECUTION
/root/helix/work/correlation/corerun -p System.Reflection.Metadata.MetadataUpdater.IsSupported=false GitHub_35821.dll ''
Gathering state for process 815 corerun
Crashing thread 0000032f signal 00000006
Writing minidump with heap to file /home/helixbot/dotnetbuild/dumps/coredump.815.dmp
Written 130756608 bytes (31923 pages) to core file
Dump successfully written
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=/root/helix/work/correlation
/root/helix/work/workitem/e/JIT/Regression/JitBlue/GitHub_35821/GitHub_35821/GitHub_35821.sh
Expected: True
Actual:   False


Stack trace
   at JIT_Regression._JitBlue_GitHub_35821_GitHub_35821_GitHub_35821_._JitBlue_GitHub_35821_GitHub_35821_GitHub_35821_sh()
```

### PR DESCRIPTION
**Pull request by kunalspathak**
Relevant discussion: https://github.com/dotnet/runtime/issues/66206#issuecomment-1059432582

Fixes: https://github.com/dotnet/runtime/issues/66565, https://github.com/dotnet/runtime/issues/66564, https://github.com/dotnet/runtime/issues/66206

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib , @BruceForstall , @tannergooding

**Comment by tannergooding** [comment]
We should also remove the `FeatureSIMD` entry at https://github.com/dotnet/runtime/blob/b6fb1d5d7c5607d040e5a958e7e638f9600fb904/src/tests/Common/testenvironment.proj#L38

and while not required for this PR, we should have an issue tracking the removal of the `COMPlus_FeatureSIMD` flag more generally (this is different from the `#if defined(FEATURE_SIMD)` macro, which needs to stay).

File: src/tests/Common/testenvironment.proj [hunk 1]
```diff
@@ -96,8 +96,7 @@
     <TestEnvironment Include="jitstress2_tiered" JitStress="2" TieredCompilation="1" />
     <TestEnvironment Include="jitstress_isas_incompletehwintrinsic" EnableIncompleteISAClass="1"  />
     <TestEnvironment Include="jitstress_isas_nohwintrinsic" EnableHWIntrinsic="0" />
-    <TestEnvironment Include="jitstress_isas_nohwintrinsic_nosimd" EnableHWIntrinsic="0" FeatureSIMD="0" />
-    <TestEnvironment Include="jitstress_isas_nosimd" FeatureSIMD="0" />
+    <TestEnvironment Include="jitstress_isas_nohwintrinsic" EnableHWIntrinsic="0" />
```
Comment (tannergooding) [inline src/tests/Common/testenvironment.proj:99]: This one is now identical to the one defined at L98

Comment (kunalspathak) [reply src/tests/Common/testenvironment.proj:99]: ah, yes.

**Comment by kunalspathak** [comment]
>  we should have an issue tracking the removal of the COMPlus_FeatureSIMD flag more generally

Updated. Let me know if I missed something.

**Comment by kunalspathak** [comment]
NativeAOT failures seems related to https://github.com/dotnet/runtime/issues/66191

**Comment by kunalspathak** [comment]
ping

**Review by tannergooding — APPROVED** [review]

**Pull request merged**